pkgs/build-support/node/fetch-pnpm-deps/pnpm-config-hook.sh

```diff
@@ -0,0 +1,44 @@
+# shellcheck shell=bash
+
+pnpmConfigHook() {
+    echo "Executing pnpmConfigHook"
+
+    if [ -n "${pnpmRoot-}" ]; then
+      pushd "$pnpmRoot"
+    fi
+
+    if [ -z "${pnpmDeps-}" ]; then
+        echo "Error: 'pnpmDeps' should be set when using pnpmConfigHook."
+        exit 1
+    fi
+
+    echo "Configuring pnpm store"
+
+    export HOME=$(mktemp -d)
+    export STORE_PATH=$(mktemp -d)
+
+    cp -Tr "$pnpmDeps" "$STORE_PATH"
+    chmod -R +w "$STORE_PATH"
```

Comment:
before copying these over, we should probably make sure the lockfile in the source and dependency derivation are the same -- similar to [npmConfigHook](https://github.com/NixOS/nixpkgs/blob/c12b90be95acc0ab9102098d34ffb6dcb52cc55b/pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh#L34)

Reply:
We don't actually ship the `pnpm-lock.yaml` in the FOD. I don't think it matters much here, as a different lock file is certainly going to cause an installation failure, as not all packages will be available.